core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java

```diff
@@ -50,17 +49,7 @@ private LoggingDecorators() {}
     public static void logWhenComplete(
             Logger logger, RequestContext ctx,
             Consumer<RequestOnlyLog> requestLogger, Consumer<RequestLog> responseLogger) {
-        final boolean isTransientService = isTransientService(ctx);
-        final CompletableFuture<RequestOnlyLog> requestCompletionFuture;
-        if (!isTransientService) {
-            requestCompletionFuture = ctx.log().whenRequestComplete().thenApply(log -> {
-                requestLogger.accept(log);
-                return log;
-            });
-        } else {
-            requestCompletionFuture = ctx.log().whenRequestComplete();
-        }
-        requestCompletionFuture.exceptionally(e -> {
+        ctx.log().whenRequestComplete().thenAccept(requestLogger).exceptionally(e -> {
```

Comment:
I feel like there are too many callbacks added here. How about just adding one callback with `handle()` and doing everything there (could catch a `Throwable`), including the `thenAccept...exceptionally` below as well?

Reply:
That's a good suggestion. Fixed. :wink: